internal/cluster/distribution/eks/eksprovider/workflow/select_volume_size_activity.go

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"emperror.dev/errors"
+)
+
+const (
+	SelectVolumeSizeActivityName = "eks-select-volume-size-activity"
+)
+
+type SelectVolumeSizeActivity struct {
+	defaultVolumeSize int
+	unsetVolumeSize   int
+}
+
+type SelectVolumeSizeActivityInput struct {
+	EKSActivityInput
+	AMIImageSize int
+}
+
+type SelectVolumeSizeActivityOutput struct {
+	VolumeSize int
+}
+
+func NewSelectVolumeSizeActivity(defaultVolumeSize, unsetVolumeSize int) (activity *SelectVolumeSizeActivity) {
+	return &SelectVolumeSizeActivity{
+		defaultVolumeSize: defaultVolumeSize,
+		unsetVolumeSize:   unsetVolumeSize,
+	}
+}
+
+func (activity *SelectVolumeSizeActivity) Execute(ctx context.Context, input SelectVolumeSizeActivityInput) (output *SelectVolumeSizeActivityOutput, err error) {
+	if input.AMIImageSize == 0 {
```

Comment:
Is this a realistic scenario? What if we just return 0? Then CloudFormation will automatically use the AMI size and we can avoid failing the process. WDYT?

Reply:
In case we assume the input comes from AWS calls I wouldn't say this is realistic, but there is no contract in the code to guarantee that so factoring in user input I would say it is a realistic scenario to receive a logically invalid value here - I would even go further and say I made a colossal mistake here by not checking the ints throughout the logic for negativity too, I'm going to go over the whole changeset and correct that.

Regarding the handling of such cases I think it's safe to handle these cases (negative or zero AMI size) uniformly with the regular positive size cases with the `explicit > default > max(unset, AMISize)` logic, I wouldn't shortcut these to `if AMISize <= 0 then return 0`, because of the unset to internal value (50 GB) logic which should prevail in this case, it gives a more consistent user experience IMO even in such edge cases.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/0e30f5dbe8ec96b1e168c848aadfd65cfc408226..4a93f8bc4925e894ff035584174682ce66d91d93).